generators/Exercises/Exercise.cs

```diff
@@ -42,40 +42,24 @@ private static void AddTestMethodUsingNamespaces(TestClass testClass)
                 testClass.UsingNamespaces.UnionWith(testMethod.UsingNamespaces);
         }
 
-        protected abstract TestMethod CreateTestMethod(TestMethodData testMethodData);
-
-        protected virtual IEnumerable<TestMethod> CreateTestMethods(CanonicalData canonicalData) 
-            => canonicalData.Cases.Select((t, i) => CreateTestMethod(canonicalData, t, i));
-        
-        protected virtual TestMethod CreateTestMethod(CanonicalData canonicalData, CanonicalDataCase canonicalDataCase, int index)
+        private static TestMethod CreateTestMethod(TestMethodData testMethodData)
         {
-            var testMethodData = CreateTestMethodData(canonicalData, canonicalDataCase, index);
-            
-            if (testMethodData.CanonicalDataCase.Expected is JObject jObject)
-                return CreateExceptionTestMethod(testMethodData);
-
-            return CreateTestMethod(testMethodData);
-        }
+            if (testMethodData.Options.ThrowExceptionWhenExpectedValueEquals(testMethodData.CanonicalDataCase.Expected))
+                return ExceptionTestMethod.Create(testMethodData);
 
-        protected virtual TestMethodData CreateTestMethodData(CanonicalData canonicalData, CanonicalDataCase canonicalDataCase, int index) 
-            => new TestMethodData
-                {
-                    CanonicalData = canonicalData,
-                    CanonicalDataCase = canonicalDataCase,
-                    Index = index,
-                    Options = CreateTestMethodOptions(canonicalData, canonicalDataCase, index)
-                };
+            if (testMethodData.Options.TestedMethodType == TestedMethodType.BooleanComparison)
```

Comment:
Minor: To future proof this you may want to have a switch with the two types existing today and then throw an exception for the default. That way if we need to add a new type in the future it's clear where it needs to be plumbed through to.